src/main/java/mate/controller/car/CreateCarController.java

```diff
@@ -0,0 +1,45 @@
+package mate.controller.car;
+
+import java.io.IOException;
+import java.util.Collections;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.model.Car;
+import mate.model.Manufacturer;
+import mate.service.CarService;
+import mate.service.ManufacturerService;
+
+@WebServlet(urlPatterns = "/cars/add")
+public class CreateCarController extends HttpServlet {
+    private static Injector injector = Injector
+            .getInstance("mate");
+    private final CarService carService = (CarService) injector
+            .getInstance(CarService.class);
+    private final ManufacturerService manufacturerService = (ManufacturerService) injector
+            .getInstance(ManufacturerService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/cars/add.jsp")
+                .forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String model = req.getParameter("model");
+        Long id = Long.valueOf(req.getParameter("manufacturerId"));

```

Comment:
```suggestion
        Long manufacturerId = Long.valueOf(req.getParameter("manufacturerId"));
```
taking into account that we create a car here, it's better to specify whose id this is

Reply:
+